src/parser.ts

```diff
@@ -492,18 +496,16 @@ export class Parser {
     if (!propsObj.valueDeclaration) {
       return {};
     }
+
     const propsType = this.checker.getTypeOfSymbolAtLocation(
       propsObj,
       propsObj.valueDeclaration
     );
-    let propertiesOfProps = propsType.getProperties();
-
-    if (!propertiesOfProps.length && propsType.isUnionOrIntersection()) {
-      propertiesOfProps = propsType.types.reduce<ts.Symbol[]>(
-        (acc, type) => [...acc, ...type.getProperties()],
-        []
-      );
-    }
+    const baseProps = propsType.getProperties();
+    const propertiesOfProps: ts.Symbol[] = propsType.isUnionOrIntersection()
+      ? // Using internal typescript API to get all properties
+        (this.checker as any).getAllPossiblePropertiesOfTypes(propsType.types)
```

Comment:
The first big think I did was use this internal TypeScript compiler API. It is way better at getting all of the types than what I did before.

Reply:
The question is how this internal API is supported across the typescript versions. Maybe it would make sense to also include those version as peer dependency for this package. Which is requested in #245 and probably make sense.  @hipstersmoothie  What do you think?

Reply:
Oh sorry, didn't noticed #248 before. Does it means that this internal API is there from version 3 onward.

Reply:
I checked an it's been present all of v3 https://www.unpkg.com/typescript@3.0.1/lib/typescript.js

At @orta's [suggestion](https://twitter.com/orta/status/1254002162092224513) I opened up [an issue](https://github.com/microsoft/TypeScript/issues/38184) to request that this API be made public